lib/api/href.rb

```diff
@@ -44,24 +67,8 @@ def ensure_prefix(str)
       result
     end
 
-    def collection
-      path_parts[version? ? 3 : 2]
-    end
-
-    def collection_id
-      path_parts[version? ? 4 : 3]
-    end
-
-    def subcollection
-      path_parts[version? ? 5 : 4]
-    end
-
-    def subcollection_id
-      path_parts[version? ? 6 : 5]
-    end
-
-    def version?
-      @version ||= !!(Api::VERSION_REGEX =~ path_parts[2])
+    def ensure_uncompressed(id)
+      Api.compressed_id?(id) ? Api.uncompress_id(id) : id
     end
 
     def path_parts
```

Comment:
(I know it's not part of this PR but now you're changing callers of this anyway)

I actually had no idea what `path_parts` would mean by it's title until I saw the code here. I feel like URL *segments* is more of a common name, which seems to be confirmed by hits on Google between those two. Maybe change this to `path_segments`?

Reply:
👍 I love `path_segments`